Content.Server/Anomaly/Components/ReagentProducerAnomalyComponent.cs

```diff
@@ -0,0 +1,129 @@
+    /// Blocked reagents for the anomaly.
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public List<ProtoId<ReagentPrototype>> BlacklistChemicals = new();
+    /// <summary>
+    /// Category of dangerous reagents for injection. Various toxins and poisons
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public List<ProtoId<ReagentPrototype>> DangerousChemicals = new();
+    /// <summary>
+    /// Category of useful reagents for injection. Medicine and other things that players WANT to get
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public List<ProtoId<ReagentPrototype>> UsefulChemicals = new();
+    /// <summary>
+    /// Category of fun reagents for injection. Glue, drugs, beer. Something that will bring fun.
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public List<ProtoId<ReagentPrototype>> FunChemicals = new();
+
+
+    /// <summary>
+    /// Increasing severity by the specified amount will cause a change of reagent.
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public float ReagentChangeStep = 0.1f;
+    /// <summary>
+    /// The next threshold beyond which the anomaly will change its reagent.
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public float NextChangeThreshold = 0.1f;
+
+    /// <summary>
+    /// Noise made when anomaly pulse.
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public SoundSpecifier ChangeSound = new SoundPathSpecifier("/Audio/Effects/waterswirl.ogg");
+
+    /// <summary>
+    /// The component will repaint the sprites of the object to match the current color of the solution,
+    /// if the RandomSprite component is hung correctly.
+    /// Ideally, this should be put into a separate component, but I suffered for 4 hours,
+    /// and nothing worked out for me. So for now it will be like this.
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public bool NeedRecolor = false;
+
+    /// <summary>
+    /// the maximum amount of reagent produced per second
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public float MaxReagentProducing = 1.5f;
+
+    /// <summary>
+    /// how much does the reagent production increase before entering the supercritical state
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public float SupecriticalReagentProducingModifier = 100f;
+
+    /// <summary>
+    ///how much anomaly produces reagent per second, with rigor and supercritical state.
+    ///
+    /// I suspect that it would be possible to do without this variable,
+    /// but I do not know how to access information about the anomaly in the Update cycle
+    /// of the component in order to calculate this value there.
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public float RealReagentProducing = 1f;
```

Comment:
There is a problem that the value of reagent production should be available in the Update cycle, from where there is no access to the component by default. Most likely it can be solved somehow, but I don't know how yet

Reply:
```
var query = EntityQueryEnumerator<..., AnomalyComponent>();
while (query.MoveNext(..., out var anomaly))
```